Source/GSThemeDrawing.m

```diff
@@ -3475,16 +3484,352 @@ - (void) drawTableViewRow: (NSInteger)rowIndex
+  hasMethod = [delegate respondsToSelector: @selector(outlineView:viewForTableColumn:item:)];
+  if (hasMethod)
+    {
+      ov = (NSOutlineView *)v;
+      outlineTableColumn = [ov outlineTableColumn];
     }
   else
     {
-      return YES;
+      hasMethod = [delegate respondsToSelector: @selector(tableView:viewForTableColumn:row:)];
+    }
+  
+  /* Determine starting column as fast as possible */
+  x_pos = NSMinX (clipRect);
+  i = 0;
+  while ((i < numberOfColumns) && (x_pos > columnOrigins[i]))
+    {
+      i++;
+    }
+  startingColumn = (i - 1);
+
+  if (startingColumn == -1)
+    startingColumn = 0;
+
+  /* Determine ending column as fast as possible */
+  x_pos = NSMaxX (clipRect);
+  while ((i < numberOfColumns) && (x_pos > columnOrigins[i]))
+    {
+      i++;
     }
+  endingColumn = (i - 1);
+
+  if (endingColumn == -1)
+    endingColumn = numberOfColumns - 1;
+
+  /* Draw the row between startingColumn and endingColumn */
+  for (i = startingColumn; i <= endingColumn; i++)
+    {
+      NSRect drawingRect = [v frameOfCellAtColumn: i
+					      row: rowIndex];
+      NSTableColumn *tb = [tableColumns objectAtIndex: i];
+      NSIndexPath *path = [NSIndexPath indexPathForItem: i
+					      inSection: rowIndex];
+      NSView *view = [v _renderedViewForPath: path];
+
+      if (ov != nil)
+	{
+	  id item = [ov itemAtRow: rowIndex];
+	  CGFloat indentationPerLevel = [ov indentationPerLevel];
+	  
+	  if (tb == outlineTableColumn)
+	    {
+	      NSImage *image = nil;
+	      NSInteger level = 0;
+	      CGFloat indentationFactor = 0.0;
+	      NSImageView *imageView = nil;
+	      NSRect imageRect = NSZeroRect;
+	      
+	      // display the correct arrow...
+	      if ([ov isItemExpanded: item])
+		{
+		  image = [NSImage imageNamed: @"common_ArrowDownH"];
+		}
+	      else
+		{
+		  image = [NSImage imageNamed: @"common_ArrowRightH"];
+		}
+	      
+	      if (![ov isExpandable: item])
+		{
+		  image = AUTORELEASE([[NSImage alloc] initWithSize: NSMakeSize(14.0,14.0)]);
+		}
+	      
+	      level = [ov levelForItem: item];
+	      indentationFactor = indentationPerLevel * level;
+	      imageView = [[NSImageView alloc] init];
+	      [imageView setImage: image];
+	      imageRect = [ov frameOfOutlineCellAtRow: rowIndex];
+	      
+	      /* Do not indent if the delegate set the image to nil. */
+	      if ([imageView image])
+		{
+		  imageRect.size.width = [image size].width;
+		  imageRect.size.height = [ov rowHeight];
+		  
+		  // Place the image...
+		  [imageView setFrame: imageRect];
+		  [ov addSubview: imageView];
+		  
+		  drawingRect.origin.x
+		    += indentationFactor + imageRect.size.width + 5;
+		  drawingRect.size.width
+		    -= indentationFactor + imageRect.size.width + 5;
+		}
+	      else
+		{
+		  drawingRect.origin.x += indentationFactor;
+		  drawingRect.size.width -= indentationFactor;
+		}
+	      
+	      RELEASE(imageView);
+	    }
+	  
+	  if (view == nil)
+	    {
+	      if (hasMethod)
```

Comment:
The variable `hasMethod` was set for both cases. You could have an NSTableViewDelegate here and that won't respond to this method.